### DISCUSSION TITLE
Communication between containers?

### DISCUSSION THREAD

**Discussion by diazwatson** [wardenenv/warden#777] [Help & Support]
I am working on a Magento2 project that requires to communicate with external services.
Those external services are running in their own container (no using warden).

If I had to work with this Magento2 project running Nginx, MySql, PHP etc directly on my mac I would just map the service containers in my host file as follow:

```
127.0.0.1 foo.baz.com
```

But using warden, what can I do to allow Magento2 to communicate with all these services in another container?
Has someone had a similar issue?

Thanks in advance

**Comment by bap14** [comment]
If the services you're working with are in a docker container, and using an image you built locally or that's publicly available, you can just add it to your `.warden/warden-env.yml` file. Then your warden environment would have access to it using the service name or other alias you give it.

**Reply by diazwatson** [reply]
Many thanks for you reply @bap14 !!!
Do you have any example of how to declare external docker containers in the `.warden/warden-env.yml` file?
Or point me in the right direction?

**Reply by bap14** [reply]
```yaml
services:
  external_service:
    image: my/custom-image:latest
```

That should be all that's needed. Then in your Magento container, you'd just point your request to `external_service` and it will find it. This is exactly how Redis, Varnish, RabbitMQ and MySQL connections work. Do a `warden env up` to start the new service.

You can double-check the whole configuration for that environment by running `warden env config` and it will output the whole `docker-compse.yml` for the environment.

**Reply by diazwatson** [reply]
But is this not going to duplicate my external service container inside my Magento group of containers?

All I need is to connect to an already running external container which is accessible in my host through `foo.baz.com`
What I need is to access `foo.baz.com` but within the Magento container.

PS: I know, I know I should have finished that docker course in Udemy 🤦‍♂️

**Reply by bap14** [reply]
@diazwatson Under the hood Warden runs a unique network for each project. That network allows communication within it. So Traefik sits as the proxy between your host and that project network.

To allow your project to connect to that one container, you need to add the external container to your project's network. You just add your project network as a network to connect to. The project network name should be your environment name suffixed with `_default`. So if your project name is `magento2` then the network name would be `magento2_default`.

If you're external service is running via docker-compose.yml file, you can just add a `networks` key to your service container definition and list `magento2_default` as one of the networks to connect to. This may mean you need to add your default network in that list as well. As an example:

```yaml
services:
  image: foo-baz-com
  neworks:
    - default
    - magento2_default
```

You can read more in the composer reference documentation: https://docs.docker.com/compose/compose-file/05-services/#networks
